ui/pages/confirmations/components/confirm/header/header.tsx

```diff
@@ -83,6 +79,22 @@ const Header = () => {
       </Box>
     </Box>
   );
+
+  // The new header includes only a heading, the advanced details toggle, and a
+  // back button if it's a wallet initiated confirmation. The default header is
+  // the original header for the redesigns and includes the sender and recipient
+  // addresses as well.
+  const isConfirmationWithNewHeader =
+    currentConfirmation?.type &&
+    CONFIRMATIONS_WITH_NEW_HEADER.includes(currentConfirmation.type);
+  const isWalletInitiated =
+    (currentConfirmation as TransactionMeta)?.origin === 'metamask';
+  if (isConfirmationWithNewHeader && isWalletInitiated) {
+    return <WalletInitiatedHeader />;
+  } else if (isConfirmationWithNewHeader && !isWalletInitiated) {
```

Comment:
Minor, we're using an `else` here but no need since we're returning in each condition.